esp/src/src-react/components/forms/CheckPermissions.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from "react";
+import { DefaultButton, Dropdown, IDropdownOption, MessageBar, MessageBarType, PrimaryButton, TextField } from "@fluentui/react";
+import { scopedLogger } from "@hpcc-js/util";
+import { useForm, Controller } from "react-hook-form";
+import nlsHPCC from "src/nlsHPCC";
+import { MessageBox } from "../../layouts/MessageBox";
+import { Users } from "../Users"
+import { Groups } from "../Groups";
+
+const logger = scopedLogger("src-react/components/forms/CheckPermissions.tsx");
+
+interface CheckPermissionsFormValues {
+    FileName: string;
+    SelectedType: string;
+    SelectedValue: string;
+}
+
+const defaultValues: CheckPermissionsFormValues = {
+    FileName: "",
+    SelectedType: "",
+    SelectedValue: "",
+};
+
+interface CheckPermissionsFormProps {
+    refreshGrid?: () => void;
+    showForm: boolean;
+    setShowForm: (_: boolean) => void;
+}
+
+export const CheckPermissionsForm: React.FunctionComponent<CheckPermissionsFormProps> = ({
+    refreshGrid,
+    showForm,
+    setShowForm
+}) => {
+
+    const { handleSubmit, control, reset, setValue, watch } = useForm<CheckPermissionsFormValues>({ defaultValues });
+
+    const [showError, setShowError] = React.useState(false);
+    const [errorMessage] = React.useState("");
+    const [userOptions, setUserOptions] = React.useState<IDropdownOption[]>([]);
+    const [groupOptions, setGroupOptions] = React.useState<IDropdownOption[]>([]);
+    const selectedType = watch("SelectedType");
+
+    React.useEffect(() => {
+        const fetchUsersAndGroups = async () => {
```

Comment:
Probably several errors in this, but I believe the inside of this useEffect here should be something like:
<pre>
React.useEffect(() => {
  WsAccess.CheckFilePermissions().then(({ FilePermissionResponse }) => {
    const { Users, Groups } = { ...FilePermissionResponse };

    setUserOptions(Users?.map(user => {
      return { key: user.username, text: user.username };
    }) ?? []);

    setGroupOptions(Groups?.map(group => {
      return { key: group.name, text: group.name };
    }) ?? []);
  }).catch(err => //etc...);
}, []);
</pre>

Where `WsAccess` would be imported from "src/WsAccess" (eg [.../components/forms/AddGroup.tsx#L6](https://github.com/hpcc-systems/HPCC-Platform/blob/master/esp/src/src-react/components/forms/AddGroup.tsx#L6))